src/ssh/cockpitsshrelay.c

```diff
@@ -622,6 +657,56 @@ set_knownhosts_file (CockpitSshData *data,
         }
     }
 
+  /* last (most expensive) fallback is to ask sssd's ssh known_hosts proxy */
+  if (!host_known && tmp_knownhost_file == NULL)
+    {
+      char port_str[10];
+      const gchar *argv [] = {
+          "sss_ssh_knownhostsproxy",
+          "--pubkey",
+          "--port", port_str,
+          host,
+          NULL
+      };
+      gchar *sout = NULL;
+      gchar *serr = NULL;
+      GError *error = NULL;
+      gint exit;
+
+      g_debug ("%s: host not known in any local file, asking sssd", data->logname);
+
+      snprintf (port_str, sizeof (port_str), "%u", port);
+
+      if (g_spawn_sync (NULL, (gchar **) argv, NULL, G_SPAWN_SEARCH_PATH, NULL,
+                        NULL, &sout, &serr, &exit, &error))
+        {
+          sout = g_strstrip (sout);
+          if (exit == 0 && strlen (sout) > 0)
+            {
+              gchar *known_hosts = prepend_host_port (host, port, sout);
+              gboolean res = write_tmp_knownhosts_file (data, known_hosts, &problem);
+              g_free (known_hosts);
+              if (!res)
+                goto out;
```

Comment:
Would `serr`, and `sout` get freed in this case?

Reply:
Looks good to me otherwise!

Reply:
Fixed.